src/pal/inc/pal.h

```diff
@@ -201,7 +201,13 @@ extern "C" {
 
 #else // _MSC_VER
 
+#if defined (_M_ARM) && !defined (DEBUG)
```

Comment:
> Please use _ARM_ instead of _M_ARM.

Basically, I though that the valid macro is _M_ARM. I will try to change the macro from _M_ARM to _ARM. 

> And why do you define it differently for DEBUG and non-DEBUG? 

I have defined it for DEBUG (-O0) and non-DEBUG (for Checked/-O2 and Release/-O3). 

> It seems it would not hurt anything.

I can not guarantee it would not hut anything because I did have enough evaluation. I would like to remove "!defined (DEBUG)" flag after evaluating the unit-test and the aging-test enough as a next step if we need.